### PR TITLE
store: fix routine/client leak when get a cached store

### LINKED ISSUE #42778: routine/client leak when get a cached store
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
we should close pd client before return cached store. and close resource on err
https://github.com/pingcap/tidb/blob/8f3328f6a335e62d9342492457226a73ee9d2537/store/driver/tikv_driver.go#L179

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #42778

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - return error after create spkv and store, check there should not have routine leak
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix routine/client leak when get a cached store
```

### CODE REVIEW

File: tests/realtikvtest/txntest/isolation_test.go [hunk 1]
```diff
@@ -17,12 +17,28 @@ package txntest
...
 
+	"github.com/pingcap/tidb/store/driver"
 	"github.com/pingcap/tidb/testkit"
 	"github.com/pingcap/tidb/tests/realtikvtest"
 	"github.com/pingcap/tidb/util"
 	"github.com/stretchr/testify/require"
+	"go.opencensus.io/stats/view"
 )
 
+func TestGetCachedStore(t *testing.T) {
```
Comment (D3Hunter) [inline tests/realtikvtest/txntest/isolation_test.go:28]: need real tikv to test it, so put it here

File: tests/realtikvtest/txntest/isolation_test.go [hunk 2]
```diff
@@ -17,12 +17,28 @@ package txntest
...
+	store1, err := d.Open(*realtikvtest.TiKVPath)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, store1.Close())
+	}()
+	store2, err := d.Open(*realtikvtest.TiKVPath)
+	require.NoError(t, err)
+	require.Equal(t, store1, store2)
+}
+
```
[the remark this answers is not in the document]
Comment (D3Hunter) [reply tests/realtikvtest/txntest/isolation_test.go:41]: already has it in `main_test.go` in this package

[the remark this answers is not in the document]
Comment (lance6716) [reply tests/realtikvtest/txntest/isolation_test.go:41]: OK, I supposed `realtikvtest.RunTestMain(m)` should have detected this leak. Maybe we can remove some `IgnoreTopFunction` now?

[the remark this answers is not in the document]
Comment (D3Hunter) [reply tests/realtikvtest/txntest/isolation_test.go:41]: they're not in the list, like `github.com/tikv/pd/client.(*client).handleResourceTokenDispatcher`

**Review by lance6716 — APPROVED** [review]

File: tests/realtikvtest/txntest/isolation_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (D3Hunter) [reply tests/realtikvtest/txntest/isolation_test.go:41]: previous, there's no test case to get the `store` twice, but in physical mode we will get one.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**